warp/src/js_exports/constellation.rs

```diff
@@ -13,3 +16,382 @@ impl ConstellationBox {
         }
     }
 }
+
+/// impl Constellation trait
+#[wasm_bindgen]
+impl ConstellationBox {
+    pub fn modified(&self) -> js_sys::Date {
+        self.inner.modified().into()
+    }
+
+    pub fn root_directory(&self) -> Directory {
+        Directory {
+            inner: self.inner.root_directory(),
+        }
+    }
+
+    pub fn max_size(&self) -> usize {
+        self.inner.max_size()
+    }
+
+    pub async fn put_buffer(&mut self, name: &str, buffer: &[u8]) -> Result<(), JsError> {
+        self.inner
+            .put_buffer(name, buffer)
+            .await
+            .map_err(|e| e.into())
+    }
+
+    pub async fn get_buffer(&self, name: &str) -> Result<JsValue, JsError> {
+        self.inner
+            .get_buffer(name)
+            .await
+            .map_err(|e| e.into())
+            .map(|ok| serde_wasm_bindgen::to_value(&ok).unwrap())
+    }
+
+    // pub async fn put_stream(&mut self, name: &str, total_size: Option<usize>, stream: BoxStream<'static, Vec<u8>>) -> Result<ConstellationProgressStream, JsError> {
+    //     self.inner.put_stream(name, total_size, stream).await.map_err(|e| e.into())
+    // }
+
+    pub async fn get_stream(&self, name: &str) -> Result<AsyncIterator, JsError> {
+        self.inner
+            .get_stream(name)
+            .await
+            .map_err(|e| e.into())
+            .map(|s| {
+                AsyncIterator::new(Box::pin(s.map(|t| {
+                    serde_wasm_bindgen::to_value(&t.map_err(|e| String::from(e.to_string())))
+                        .unwrap()
+                })))
+            })
+    }
+
+    pub async fn remove(&mut self, name: &str, recursive: bool) -> Result<(), JsError> {
+        self.inner
+            .remove(name, recursive)
+            .await
+            .map_err(|e| e.into())
+    }
+
+    pub async fn rename(&mut self, current: &str, new: &str) -> Result<(), JsError> {
+        self.inner.rename(current, new).await.map_err(|e| e.into())
+    }
+
+    pub async fn create_directory(&mut self, name: &str, recursive: bool) -> Result<(), JsError> {
+        self.inner
+            .create_directory(name, recursive)
+            .await
+            .map_err(|e| e.into())
+    }
+
+    pub async fn sync_ref(&mut self, path: &str) -> Result<(), JsError> {
+        self.inner.sync_ref(path).await.map_err(|e| e.into())
+    }
+
+    pub fn set_path(&mut self, path: String) {
+        self.inner.set_path(path.into())
+    }
+
+    pub fn get_path(&self) -> String {
+        self.inner.get_path().to_str().unwrap().into()
+    }
+}
+
+#[wasm_bindgen]
+#[derive(Clone)]
+pub struct Directory {
+    inner: constellation::directory::Directory,
+}
```

Comment:
Any reason why we couldnt mark `Directory` in `constellation` with `#[wasm_bindgen]` and add the functions there to point to the respective functions?